ksqldb-serde/src/main/java/io/confluent/ksql/serde/connect/ConnectFormatSchemaTranslator.java

```diff
@@ -59,16 +59,29 @@ public String name() {
   }
 
   @Override
-  public List<SimpleColumn> toColumns(final ParsedSchema schema, final boolean isKey) {
+  public List<SimpleColumn> toColumns(
+      final ParsedSchema schema,
+      final boolean isKey,
+      final boolean unwrapSingle) {
     Schema connectSchema = connectSrTranslator.toConnectSchema(schema);
 
-    if (connectSchema.type() != Type.STRUCT) {
+    if (connectSchema.type() != Type.STRUCT || unwrapSingle) {
```

Comment:
I don't think we need to check both.   Just checking the serde features is enough.  So I think this should become:

```java
 @Override
  public List<SimpleColumn> toColumns(
      final ParsedSchema schema,
      final SerdeFeatures serdeFeatures,
      final boolean isKey
  ) {
    SerdeUtils.throwOnUnsupportedFeatures(serdeFeatures, format.supportedFeatures());

    Schema connectSchema = connectSrTranslator.toConnectSchema(schema);

    if (serdeFeatures.enabled(SerdeFeature.UNWRAP_SINGLES)) {
      connectSchema = SerdeUtils.wrapSingle(connectSchema, isKey);
    }

    if (connectSchema.type() != Type.STRUCT) {
      if (isKey) {
        throw new IllegalStateException("Key schemas are always unwrapped.");
      }
      
      throw new KsqlException("Schema returned from schema registry is anonymous type. "
          + "To use this schema with ksqlDB, set '" + CommonCreateConfigs.WRAP_SINGLE_VALUE
          + "=false' in the WITH clause properties.");
    }

    final Schema rowSchema = connectKsqlTranslator.apply(connectSchema);

    return rowSchema.fields().stream()
        .map(ConnectFormatSchemaTranslator::toColumn)
        .collect(Collectors.toList());
  }
```

Reply:
Much cleaner! Thanks for the suggestion.

Wanted to check that we're OK removing the exception that's thrown if an unwrapped schema is encountered for a format that does not support unwrapping ([link](https://github.com/confluentinc/ksql/blob/a510ceef8dbad9be79f80bc9dff6e541ac580c5d/ksqldb-serde/src/main/java/io/confluent/ksql/serde/connect/ConnectFormatSchemaTranslator.java#L66-L70)) -- it makes sense to me that we're removing this since the only way we'd end up in this situation is if ksqlDB designated a format the supports unwrapping as not supporting it, but maybe it's better to be defensive and leave the check in?